sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ServiceBusProcessor.cs

```diff
@@ -960,22 +959,47 @@ public void UpdateConcurrency(int maxConcurrentCalls)
             }
         }
 
-        internal void UpdateConcurrency(int maxConcurrentSessions, int maxConcurrentCallsPerSession)
+        internal void UpdateConcurrency(int maxConcurrentSessions, int maxConcurrentCallsPerSession, int? maxConcurrentCallsAcrossAllSessions = default)
         {
             Argument.AssertAtLeast(maxConcurrentSessions, 1, nameof(maxConcurrentSessions));
             Argument.AssertAtLeast(maxConcurrentCallsPerSession, 1, nameof(maxConcurrentCallsPerSession));
 
+            if (maxConcurrentCallsAcrossAllSessions.HasValue)
+            {
+                Argument.AssertAtLeast(maxConcurrentCallsAcrossAllSessions.Value, 1, nameof(maxConcurrentCallsAcrossAllSessions));
+            }
+
             lock (_maxConcurrencySyncLock)
             {
-                _maxConcurrentCalls = _sessionIds.Length > 0
-                    ? Math.Min(_sessionIds.Length, maxConcurrentSessions)
-                    : maxConcurrentSessions * maxConcurrentCallsPerSession;
                 _maxConcurrentSessions = maxConcurrentSessions;
                 _maxConcurrentCallsPerSession = maxConcurrentCallsPerSession;
+
+                SetMaxConcurrentCallsAcrossSessions(maxConcurrentCallsAcrossAllSessions);
                 WakeLoop();
             }
         }
 
+        private void SetMaxConcurrentCallsAcrossSessions(int? maxConcurrentCallsAcrossAllSessions)
+        {
+            int calculatedMaxConcurrentCalls = _sessionIds.Length > 0
+                ? Math.Min(_sessionIds.Length, _maxConcurrentSessions)
+                : _maxConcurrentSessions * _maxConcurrentCallsPerSession;
+            if (maxConcurrentCallsAcrossAllSessions.HasValue)
+            {
+                if (maxConcurrentCallsAcrossAllSessions.Value > calculatedMaxConcurrentCalls)
+                {
+                    throw new ArgumentOutOfRangeException(Resources.InvalidMaxConcurrentCalls);
+                }
+                _maxConcurrentCalls = maxConcurrentCallsAcrossAllSessions.Value;
+            }
+            else
+            {
+                _maxConcurrentCalls = _sessionIds.Length > 0
```

Comment:
Rather than repeat this logic again, just use calculatedMaxConcurrentCalls from above?